src/main/java/org/openrewrite/java/logging/slf4j/Slf4jLogShouldBeConstant.java

```diff
@@ -84,18 +86,17 @@ public J visitMethodInvocation(J.MethodInvocation method, ExecutionContext execu
                         if (STRING_FORMAT.matches(args.get(0))) {
                             J.MethodInvocation stringFormat = (J.MethodInvocation) args.get(0);
 
-                            if (stringFormat.getArguments() == null ||
-                                    stringFormat.getArguments().size() <= 1 ||
+                            if (stringFormat.getArguments().size() <= 1 ||
                                     !CompleteExceptionLogging.isStringLiteral(stringFormat.getArguments().get(0))
                             ) {
                                 return method;
                             }
 
-                            String strFormat = ((J.Literal) stringFormat.getArguments().get(0)).getValue().toString();
-                            if (containsIndexFormatSpecifier(strFormat) || containsCombinedFormatSpecifiers(strFormat)) {
+                            String strFormat = String.valueOf(((J.Literal) stringFormat.getArguments().get(0)).getValue());
+                            if (StringUtils.isBlank(strFormat) || containsIndexFormatSpecifier(strFormat) || containsCombinedFormatSpecifiers(strFormat)) {
```

Comment:
adding this check here makes the isEmptyOrNull() redundant in the other methods, however, because they are methods I think having them self contained with these checks still makes sense.

Reply:
I have a slight worry about the potential failure modes here; before if `getValue()` was ever to return `null` then we would fail clearly  with an NPE; I've heard no such complaints, but still, it's predictable and safe with regards to unintentional code changes.

With `String.valueOf(String)` we instead effectively have `(obj == null) ? "null" : obj.toString();`, which might return `"null"` as a string literal. That literal is not caught by the `StringUtils.isBlank(strFormat)` conditional added here, and I'm hesitant to add special handling for it. That `"null"` could then end up as the format string, which could result in unsafe accidental code changes, which could be missed when running this recipe at scale.

I propose we keep the original may-throw-npe over the may-replace-format-string-with-"null". Do you agree?
```suggestion
                            String strFormat = ((J.Literal) stringFormat.getArguments().get(0)).getValue().toString();
                            if (containsIndexFormatSpecifier(strFormat) || containsCombinedFormatSpecifiers(strFormat)) {
```

Reply:
Before these lines there is a check if the first arg is a string literal, which is very hard (but I believe not impossible) to make null. So this was mostly to make intellij happy.

However, hypothetically, if this were to be null, indeed we would not early exit, but is that so bad?
This would only happen if there is already a null being passed into the log line, all we would do is just rewrite it (which would fail miserably because the rest of the code also expects a literal).

But, I am not at all attached to this change, and if "this is the way" that you normally do it, consistency is more important than the above arguments.

Reply:
Ah, I see there is a flaw in my reasoning, we would rewrite null -> "null" indeed, that is not the same at all.
So lets change it back.

Still I would leave the isBlank for the early exit because all the subsequent checks are redundant. Do you agree?

Reply:
Again, bad reasoning, the subsequent checks might be redundant, but rewriting
`log.info(String.format("     "));` -> `log.info("     ");` is still sort of "useful"

Reply:
But that does not work, because we check if there is at least 2 arguments passed to `String.format()`
Why would we not want to rewrite if there are no args passed to the string format? This could indicate that the author forgot to add them, but it could also be the case that the author did pass some but not all args.

Reply:
I made another comment, please treat it as a separate suggestion, as I might be missing something important here.

Reply:
I like your alternative about being explicit about the non-null. And appreciate the additional tests to rewrite single arg string format and empty string. I'd say we're done here, thanks!